src/clj/puppetlabs/services/master/master_service.clj

```diff
@@ -19,8 +19,8 @@
          localcacert (get-in config [:puppet-server :localcacert])
          settings    (ca/config->master-settings config)]
 
-     (retrieve-ca-cert! localcacert)
      (initialize-master-ssl! settings certname)
+     (retrieve-ca-cert! localcacert)
```

Comment:
> initialize-master-ssl! would then throw a partial state error because the localcacert file already exists but the hostprivkey and hostcert do not.

Ahh, right, that scenario makes sense now. And yes, that would necessitate this change.

Unfortunately, this doesn't work for the inverse scenario: the hostprivkey and hostcert exist, but the localcacert does not. This is a valid and possible state, and in this case the master won't be able to start due to the partial-state error.

After discussing this IRL with @camlow325 I think it's best to not change anything about the required files for now considering this wasn't strictly necessary to begin with.